### PR TITLE
Add SwiftPM Support

### PR DESCRIPTION
**Pull request by JoeMatt**
Added a Package.swift to support iOS, tvOS and macCatalyst.

Also supports tests.

No warnings or errors. Added some minor changes for deprecations and warnings such as `class` to `AnyObject`.

### CODE REVIEW

**Review by AppDevGuy — APPROVED** [review]
Looks good, thanks for adding.

**Comment by AppDevGuy** [comment]
Build is failing because it's attempting to use an outdated version of Xcode. I will have to upgrade this package to get it working, might be a weekend thing.

**Comment by JoeMatt** [comment]
> Build is failing because it's attempting to use an outdated version of Xcode. I will have to upgrade this package to get it working, might be a weekend thing.

I pushed an update that I think will fix that.

- Lowered SPM version to 5.3
- tvOS only compiles TextToSpeech, not SpeechToText
- Replace import `UIKit` with `Foundation`, and conditionally compile `UIImage* tag` support. (For possibly supporting macOS or other targets that don't have UIKit but have the other requirements)
- Loading bundle uses `Bundle.module` if `SWIFT_PACKAGE`.

**Comment by JoeMatt** [comment]
Oh. I see you meant the gh workflow. I can take a look at that real quick.

File: OSSSpeechKit/Classes/OSSSpeechUtility.swift [hunk 1]
```diff
@@ -72,33 +73,39 @@ public class OSSSpeechUtility: NSObject {
...
-    /// Will return the Bundle for the SDK if it can be found.
-    static func getResourcesBundle() -> Bundle? {
-        let bundle = Bundle(for: OSSSpeech.self)
-        guard let resourcesBundleUrl = bundle.resourceURL?.appendingPathComponent("OSSSpeechKit.bundle") else {
-            return nil
-        }
-        return Bundle(url: resourcesBundleUrl)
-    }
+	/// Will return the Bundle for the SDK if it can be found.
+	static func getResourcesBundle() -> Bundle? {
```
Comment (AppDevGuy) [inline OSSSpeechKit/Classes/OSSSpeechUtility.swift:81]: Unless I am not mistaken, indentation here has increased, which is probably due to the `#if` statement, Xcode always messes up indentation when implementing those macros.

File: OSSSpeechKit/Classes/OSSVoice.swift [hunk 2]
```diff
@@ -226,12 +229,14 @@ public enum OSSVoiceEnum: String, CaseIterable {
     /// You can supply your own flag image, provided is has the same name (.rawValue) as the image in the pod assets.
     ///
     /// If no image is found in the application bundle, the image from the SDK bundle will be provided.
+	#if canImport(UIKit)
```
Comment (AppDevGuy) [inline OSSSpeechKit/Classes/OSSVoice.swift:232]: Indentation is incorrect on the `#if` here as well.

**Review by AppDevGuy — COMMENTED** [review]
Great catches and effort on identifying the tvOS issue for the Swift Package. Just a couple of indentation issues.

File: OSSSpeechKit/Classes/OSSVoice.swift (hunk 2 above)
Comment (JoeMatt) [reply OSSSpeechKit/Classes/OSSVoice.swift:234]: Fixing. I've seen it both ways though. Swift-format allows indenting though Xcode by default doens't indent them.

**Comment by JoeMatt** [comment]
Pushed updates.

- Fixed? xcode gh action
- Added swift pm gh action
- Added macOS support (this is to allow swift build/text from cli without having to remember how to tell it to use iOS, though it works on mac now too so why not).
- Fixed indenting issues
- Fixed broken force unwrap in test

**Comment by JoeMatt** [comment]
One last push after SwiftLint autocorrect.

**Comment by AppDevGuy** [comment]
Tests failed which I believe I know the reason why - the `let archiver = NSKeyedUnarchiver(forReadingWith: Data())` whic you replaced - I went down this rabbit hole 2 years or more ago now when it was first deprecated but it seems Apple still haven't fixed the issue.

**Comment by AppDevGuy** [comment]
Is there any chance you can rebase this on `v0.3.3_develop` branch? If so, I will merge it in there, then fix any of the small issues and do a release.

**Comment by JoeMatt** [comment]
> Is there any chance you can rebase this on `v0.3.3_develop` branch? If so, I will merge it in there, then fix any of the small issues and do a release.

Done

**Comment by AppDevGuy** [comment]
Thanks @JoeMatt - I will fix up a couple of things now and aim to release a new version soon.

**Pull request merged**